### PR TITLE
DDF-04941 Sharing breaks if the sharee's email contains capital letters

### PR DESCRIPTION
**Pull request by rymach**
reviewers may want to view the changes from one commit at a time

#### Who is reviewing it? 

@leo-sakh 
@mojogitoverhere 
@gjvera 
@Bdthomson 
#### Select relevant component teams: 

@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.

@adimka
@bdeining

#### How should this be tested?

see issue
#### Any background context you want to provide?
no
#### What are the relevant tickets?
For GH Issues:
Fixes: #4941 
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by rymach** [comment]
build now

**Comment by bdeining** [comment]
please fill out the template

**Comment by rymach** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/container/sharing/sharing.tsx [hunk 1]
```diff
@@ -128,39 +128,58 @@ export class Sharing extends React.Component<Props, State> {
     const loadingView = new LoadingView()
     this.attemptSave(attributes, usersToUnsubscribe)
       .then(() => {
```
Comment (leo-sakh) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/container/sharing/sharing.tsx]: ✏️ While I'm not opposed to this here, I think the `await` with `try/catch` would be easier to read.
```
try {
   await this.attemptSave(...)
   announcement.announce(...)
} catch (error) {
   if(error.message === ...) {
      ...
   }
}
```

**Comment by mojogitoverhere** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/container/sharing/sharing.tsx [hunk 2]
```diff
@@ -128,39 +128,58 @@ export class Sharing extends React.Component<Props, State> {
...
   attemptSave = async (attributes: any, usersToUnsubscribe: String[]) => {
     const currWorkspace = await this.fetchWorkspace(this.props.id)
-    if (
-      currWorkspace['metacard.modified'] ===
-      this.state.previousWorkspace['metacard.modified']
-    ) {
+    if (currWorkspace['metacard.modified'] === this.state.modified) {
       await this.doSave(attributes)
       await this.unsubscribeUsers(usersToUnsubscribe)
       const newWorkspace = await this.fetchWorkspace(this.props.id)
```
Comment (gjvera) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/container/sharing/sharing.tsx]: Should we update this to be `newMetacard` and the function to be `fetchMetacard` to keep in line with all the other changes being made?

Comment (rymach) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/container/sharing/sharing.tsx]: good call

**Comment by rymach** [comment]
build now

**Comment by rymach** [comment]
build now

**Comment by gjvera** [comment]
### 🎉 Hero Successful 🎉 
Shared search forms with other users using all caps for their emails
sharing worked as expected
tried this with workspaces, search forms, and result forms
tried emails with mixes of upper and lowercase letters

LGTM!!

**Comment by gjvera** [comment]
build now

**Pull request merged**